### PR TITLE
Feature/add component tests

### PR DESCRIPTION
**Pull request by andre-urbani**
### What

Add component tests - https://trello.com/c/OFvMAEX3/5182-dp-cantabular-csv-exporter-add-component-test

### How to review

Pull branch and run `make test-component` to confirm that all tests pass

### Who can review

Anyone

### CODE REVIEW

**Review by cookel2 — COMMENTED** [review]
The tests ran successfully for me but they seemed to be running services that perhaps ought to be mocked instead. My understanding is that component tests should only be running the service that they are testing; other services should be mocked up.

**Comment by ChrysmOre** [comment]
> The tests ran successfully for me but they seemed to be running services that perhaps ought to be mocked instead. My understanding is that component tests should only be running the service that they are testing; other services should be mocked up.

It's only external dependencies that are running in the containers not other ONS services, and this was a conscious decision. With the Cantabular services we've been using real instances of Kafka (and here mino too) too as we thought it would be more meaningful.. It's since developed that these tests will form the basis of upcoming integration tests we add, so at that time we may revert to mocking everything for component tests, but now this is fine.

File: features/steps/steps.go [hunk 1]
```diff
@@ -2,84 +2,213 @@ package steps
...
 
+const testETag = "13c7791bafdbaaf5e6660754feb1a58cd6aaa892"
+
+// InstanceCompleteEventsTimeout is the maximum time that the testing producer will wait
+// for DimensionComplete events to be produced before failing the test
+var InstanceCompleteEventsTimeout = time.Second
+
+// RegisterSteps maps the human-readable regular expressions to their corresponding funcs
 func (c *Component) RegisterSteps(ctx *godog.ScenarioContext) {
+	ctx.Step(`^the following response is available from Cantabular from the codebook "([^"]*)" and query "([^"]*)":$`, c.theFollowingQueryResponseIsAvailable)
```
Comment (DavidSubiros) [inline features/steps/steps.go:31]: if we implement the changes in the other comment, we should modify this accordingly

**Review by DavidSubiros — COMMENTED** [review]
- A couple of minor comments to prevent misleading query in test description (in fact, the query is always the same and not configurable by component tests)
- There will be conflicts with the latest develop, please feel free to ask about that (I did the latest changes)

File: features/compose/.env [hunk 2]
```diff
@@ -0,0 +1,3 @@
+COMPOSE_FILE=kafka-cluster.yml:dp-cantabular-csv-exporter.yml:minio.yml
+COMPOSE_PATH_SEPARATOR=:
+COMPOSE_PROJECT_NAME=dp-cantabular-csv-exporter-features
```
Comment (cookel2) [inline features/compose/.env:3]: new line needed here

File: go.mod [hunk 3]
```diff
@@ -12,14 +12,16 @@ require (
 	github.com/ONSdigital/dp-net v1.0.12
```
Comment (cookel2) [inline go.mod:12]: The latest version of dp-component-test is now 0.5.0 I've been told.

**Review by cookel2 — COMMENTED** [review]
I've added a couple of minor comments.

**Review by DavidSubiros — APPROVED** [review]
LGTM and comments have been addressed.
Also, thanks for fixing the component tests for my latest changes!

**Pull request merged**